decidim-system/app/commands/decidim/system/create_oauth_application.rb

```diff
@@ -23,7 +23,7 @@ def call
           organization_logo: @form.organization_logo,
           redirect_uri: @form.redirect_uri,
           scopes: "public",
-          organization: @form.current_organization
+          organization: Decidim::Organization.find_by(name: @form.organization_name)
```

Comment:
It seems to me that this is put very much trust in the user writing down the organization name correctly.
I think in this case the Form class should return the organization object or return invalid if not found.